cmd/erasure-decode.go

```diff
@@ -156,8 +207,9 @@ func (err *errDecodeHealRequired) Unwrap() error {
 }
 
 // Decode reads from readers, reconstructs data if needed and writes the data to the writer.
-func (e Erasure) Decode(ctx context.Context, writer io.Writer, readers []io.ReaderAt, offset, length, totalLength int64) error {
-	healRequired, err := e.decode(ctx, writer, readers, offset, length, totalLength)
+// A set of preferred drives can be supplied. In that case they will be used and the data reconstructed.
+func (e Erasure) Decode(ctx context.Context, writer io.Writer, readers []io.ReaderAt, offset, length, totalLength int64, prefer []bool) error {
```

Comment:
preference can be decided in this function itself instead of passing it from the caller since we always prefer local disks.

Reply:
@krishnasrinivas How? The only thing passed are `readers` which is an `io.ReaderAt`?

Either way, I would rather leave the flexibility to the caller. There could be cases while healing for instance where the caller has a better idea of the preferred.

Reply:
Furthermore we could extend it to prefer hosts that we think will be faster down the line.

Reply:
The reasoning is @krishnasrinivas preferred is sort of configurable , current code is symmetric in all situations @klauspost is trying to bring in situations where it may be possible to choose that as a policy if needed.